lib/collector/serverless.js

```diff
@@ -258,6 +285,10 @@ class ServerlessCollector {
       payload
     ]) + '\n'
 
+    if (this.shouldUsePipe()) {
+      return this.flushToPipe(serializedPayload)
+    }
```

Comment:
On cases where the pipe fails (should only be once due to error caching) we'll serialize the payload twice.

Prob not a big deal but we could potential restructure things to avoid that. Fine if you want to leave things.

This prob isn't the best code but a quick idea of what could be done...

```js
const didFlushToPipe = this.shouldUsePipe() && this.flushToPipe(serializedPayload) // make flushToPipe return bool / not execute retry itself

if (!didFlushToPipe) {
  this.flushToStdOut(serializedPayload) // move stdout code into helper function
}
```